test/abc/sound/SequencePlayerTest.java

```diff
@@ -1,38 +1,125 @@
+			player.addNote(new Pitch('E').toMidiNote(), 128, 4);
+			player.addNote(new Pitch('C').toMidiNote(), 132, 4);
+			player.addNote(new Pitch('C').toMidiNote(), 136, 4);
+			player.addNote(new Pitch('C').toMidiNote(), 140, 4);
+
+			player.addNote(new Pitch('G').toMidiNote(), 144, 9);
+			player.addNote(new Pitch('F').toMidiNote(), 153, 3);
+			player.addNote(new Pitch('E').toMidiNote(), 156, 9);
+			player.addNote(new Pitch('D').toMidiNote(), 165, 3);
+			player.addNote(new Pitch('C').toMidiNote(), 168, 24);
+
+			System.out.println(player);
+
+			player.play();
+			Thread.sleep(10000);
+
+		} catch (MidiUnavailableException midiUe) {
+			midiUe.printStackTrace();
+		} catch (InvalidMidiDataException invalidMidiDe) {
+			invalidMidiDe.printStackTrace();
+		} catch (InterruptedException intEx) {
+			intEx.printStackTrace();
+		}
+	}
+
+	@Test
+	public void piece2() {
 		try {
-			CharStream input = fromFileName(fileName);
-			AbcLexer lexer = new AbcLexer(input);
-			CommonTokenStream tokens = new CommonTokenStream(lexer);
-			AbcParser parser = new AbcParser(tokens);
-			ParseTree tree = parser.abc_tune();
-			// TODO: Implement custom walker, parse tree into POJO, play notes using SquencePlayer
-			ParseTreeWalker walker = new ParseTreeWalker();
-			AbcBaseListener listener = new AbcBaseListener();
-			walker.walk(listener, tree);
-		} catch (IOException e) {
-			e.printStackTrace();
+			SequencePlayer player = new SequencePlayer(200, 12);
+
+			player.addNote(new Pitch('E').transpose(1).toMidiNote(), 0, 6);
+			player.addNote(new Pitch('F').transpose(12).toMidiNote(), 0, 6);
+			player.addNote(new Pitch('E').transpose(1).toMidiNote(), 6, 6);
+			player.addNote(new Pitch('F').transpose(12).toMidiNote(), 6, 6);
+
+			player.addNote(new Pitch('E').transpose(1).toMidiNote(), 18, 6);
+			player.addNote(new Pitch('F').transpose(12).toMidiNote(), 18, 6);
+
+			player.addNote(new Pitch('C').transpose(1).toMidiNote(), 30, 6);
+			player.addNote(new Pitch('F').transpose(12).toMidiNote(), 30, 6);
+
+			player.addNote(new Pitch('E').transpose(1).toMidiNote(), 36, 12);
+			player.addNote(new Pitch('F').transpose(12).toMidiNote(), 36, 12);
+
+			player.addNote(new Pitch('G').toMidiNote(), 48, 12);
+			player.addNote(new Pitch('B').toMidiNote(), 48, 12);
+			player.addNote(new Pitch('G').transpose(12).toMidiNote(), 48, 12);
+
+			player.addNote(new Pitch('G').toMidiNote(), 72, 12);
+
+			player.addNote(new Pitch('C').transpose(12).toMidiNote(), 96, 18);
+			player.addNote(new Pitch('G').toMidiNote(), 114, 6);
+
+			player.addNote(new Pitch('E').toMidiNote(), 132, 12);
+
+			player.addNote(new Pitch('E').toMidiNote(), 144, 6);
+			player.addNote(new Pitch('A').toMidiNote(), 150, 12);
+			player.addNote(new Pitch('B').toMidiNote(), 162, 12);
+			player.addNote(new Pitch('B').transpose(-1).toMidiNote(), 174, 6);
+			player.addNote(new Pitch('A').toMidiNote(), 180, 12);
+
+			player.addNote(new Pitch('G').toMidiNote(), 192, 8);
```

Comment:
See, here you have all 3 notes in the tuplet playing at the same `startTick`.

Reply:
Yeah. I just changed those to different startTicks and it doesn't sound well so I had to keep it as same.